eventbus.go

```diff
@@ -35,8 +35,18 @@ func (e EventBus) UnSubscribe(eventName string, ch DataChannel) {
 
 	for idx, subscriber := range subscribers {
 		if subscriber == ch {
-			e.events[eventName] = append(subscribers[:idx], subscribers[idx+1:]...)
-			return
+			// the order of subscribers does not matter
+			subscribers[idx], subscribers[len(subscribers)-1] = subscribers[len(subscribers)-1], nil
+			e.events[eventName] = subscribers[:len(subscribers)-1]
```

Comment:
We have to remove the subscriber matching the channel: ch

Reply:
There I'm just changing the index of the channel: ch with the last channel, then resizing the slice by removing last element(which now is channel: ch)

Reply:
@arberiii Sorry, I missed that. I recommend implementing a swap function to improve its readability.